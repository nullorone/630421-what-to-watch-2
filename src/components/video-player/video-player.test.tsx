import * as React from "react";
import renderer from "react-test-renderer";
import VideoPlayer from "./video-player";

describe(`Make snapshot`, () => {
  const initProps = {
    video: {
      link: {
        mp4: `New Year`,
        webm: `Christmas`,
      },
      poster: `./img/Holly`,
    },
    isMuted: true,
  };

  const createNodeMock = (element) => {
    return element.type === `video`
      ? {
        play: () => {},
        pause: () => {},
      }
      : null;
  };

  it(`Get snapshot component`, () => {
    const tree = renderer
      .create(
          <VideoPlayer
            video={initProps.video}
          />,
          {createNodeMock}
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
