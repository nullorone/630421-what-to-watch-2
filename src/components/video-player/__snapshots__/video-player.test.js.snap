// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Make snapshot Get snapshot component 1`] = `
<video
  muted={true}
  poster="./img/Holly"
  style={
    Object {
      "height": "100%",
      "width": "100%",
    }
  }
>
  <source
    src="New Year"
    type="video/mp4"
  />
  <source
    src="Christmas"
    type="video/webm"
  />
</video>
`;
